heron/schedulers/src/java/com/twitter/heron/scheduler/nomad/NomadScheduler.java

```diff
@@ -310,8 +310,10 @@ Task getTask(String taskName, int containerIndex,
         throw new RuntimeException(msg, e);
       }
       Map<String, String> metaData = jobActual.getMeta();
-      if (metaData.get(NomadConstants.NOMAD_TOPOLOGY_NAME).equals(topologyName)) {
-        ret.add(jobActual);
+      if (metaData != null) {
+        if (metaData.get(NomadConstants.NOMAD_TOPOLOGY_NAME).equals(topologyName)) {
```

Comment:
could metaData.get here ever return null?

Reply:
Yes if there is some other nomad job running that does not have this user defined meta data set e.g. API server